documentation/src/docs/streamCuts.md

```diff
@@ -0,0 +1,87 @@
+<!--
+Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+-->
+# Working with Pravega: StreamCuts
+
+This section describes about StreamCuts and how it can be used with streaming clients and batch clients.
+Pre-requisites: You should be familiar with Pravega Concepts (see Pravega Concepts).
+
+## Definition
+
+StreamCut represents a consistent position in the stream. It essentially contains a set of segment
+and offset pairs for a single stream which represents the complete keyspace. The offset always points to
+the event boundary and hence there will be no offset pointing to an incomplete event.
+
+Given that data can be continuously added to a stream, the streamcut representing the tail of the
+stream(with the newest event) is an ever changing one. Similarly the stream cut representing the
+head of the stream(with the oldest event) is an ever changing one as the stream retention policy
+could truncate the stream. ```StreamCut.UNBOUNDED``` is used to represent such a position in the stream and
+the user can use it to specify this ever changing stream position (both head and tail of the stream).
+
+It should be noted that streamcuts obtained using the streaming client and batch client can be used
+interchangeably.
+
+## StreamCut with Streaming clients.
+
+StreamCut(s) can be obtained from a Reader group using the following api.
+```io.pravega.client.stream.ReaderGroup.getStreamCuts ```. This api returns a ```Map<Stream, StreamCut>``` which
+contains a StreamCut for every stream managed by the ReaderGroup.
+
+A StreamCut can be used to configure a ReaderGroup to enable bounded processing of a Stream. The start
+and/or end StreamCut of a Stream can be passed as part of the ReaderGroup configuration. The below example
+shows the different ways to use StreamCuts as part of the reader group configuration.
+
+```java
+/*
+ * The below ReaderGroup configuration ensures that the readers belonging to
+ * the ReaderGroup read events from
+ *   - Stream "s1" from startStreamCut1 (representing the oldest event) upto
+          endStreamCut1 (representing the newest event)
+ *   - Stream "s2" from startStreamCut2 upto the tail of the stream, this is similar to using StreamCut.UNBOUNDED
+ *        for endStreamCut.
+ *   - Stream "s3" from the current head of the stream upto endStreamCut2
+ *   - Stream "s4" from the current head of the stream upto the tail of the stream.
+ */
+ReaderGroupConfig.builder()
+                .disableAutomaticCheckpoints()
```

Comment:
Is the call to `disableAutomaticCheckpoints()` mandatory? If not, it may introduce unnecessary complexity.

Reply:
removed it.